### DISCUSSION TITLE
Controlled React Select with conditional fields

### DISCUSSION THREAD

**Discussion by zeyuri** [react-hook-form/react-hook-form#3023] [Q&A] [answered]
Hi,

I'm facing a little bug in my app where I have two React Select inputs that are conditionals: they depend on a third input value that I'm watching.

When I change the watched input, the selected value on any of those conditional inputs get persisted.

I've made a sandbox showing the exact same behavior [here](https://codesandbox.io/s/react-hook-form-conditional-fields-react-select-6egtm)

**Comment by bluebill1049** [comment, marked as answer]
what's the steps to reproduce?

**Reply by zeyuri** [reply]
1. Select anything on the first input "Ice Cream Flavor"
2. Select "cookies" on the second input "Extras"
3. Select "chocolate" on the third input "Cookies"
4. Change the select value on the second input "Extras" to "Fresh Fruits"
5. The chocolate value of the cookies input is selected on the now "Fresh Fruits" input.

**Reply by bluebill1049** [reply]
<img width="587" alt="Screen Shot 2020-09-30 at 9 50 42 am" src="https://user-images.githubusercontent.com/10513364/94628064-6852b200-0302-11eb-91d7-51929fd791f9.png">

sorry still don't get the issue clearly here... but I sort of get the problem:  try to get a key to those components that you are toggling.

https://react-hook-form.com/faqs#Whyisdefaultvaluenotchangingcorrectlywithternaryoperator

**Reply by zeyuri** [reply]
the key was the key haha, thanks
